compose/config/config.py

```diff
@@ -345,14 +345,14 @@ def validate_extended_service_dict(service_dict, filename, service):
                 "%s services with 'net: container' cannot be extended" % error_prefix)
 
 
-def process_container_options(service_dict, working_dir=None):
-    service_dict = service_dict.copy()
+def process_container_options(working_dir, service_dict):
+    service_dict = dict(service_dict)
```

Comment:
I think they are able the same thing. They would both make a shallow copy, one using the constructor and another using the copy method.  I wasn't able to find much about the differences.

I think for the case where `service_dict` is actually a `dict` they would be equivalent, if `service_dict` is `UserDict`, `OrderedDict`, or something else, it would force it to be a regular `dict` using this method, and would preserve the original type using `copy()`.

In practice it probably doesn't make any difference.  I think I only changed it because I'm used to writing `dict()`, but I can change it back.